### ISSUE TITLE
replaceWith created nodes will crash on finding _autoprefixerPrefix

### ISSUE THREAD

**Issue by ksjogo** [postcss/autoprefixer#1234]
I have a custom plugin creating some grid rules from a domain specific language, e.g.
```  rule = rule.replaceWith({
                prop: 'grid-template',
                value: result
            }, {
                    prop: 'width',
                    value: 'calc(100% - 2 * var(--gap))',
                    source: rule.source
                }, {
                    prop: 'max-width',
                    value: `calc((100vh - 2 * var(--gap)) * ${RowI * 16} / ${ColL[0] * 9} - 2 * var(--gap))`,
                    source: rule.source
                }, {
                    prop: 'grid-gap',
                    value: 'var(--gap)',
                    source: rule.source
                }, {
                    selector: '.cell:nth-child(1)',
                    nodes: [postcss.decl({
                        prop: 'grid-area',
                        value: 'a1'
                    })]
                }
            );
```

This runs fine, till autoprefixer is trying to find the parent of the 'grid-area' rule, as for some reason it thinks it is undefined, which will result in a 
`TypeError: Cannot read property '_autoprefixerPrefix' of undefined`
Stopping postcss before autoprefixer, saving, and then just passing this to autoprefixer works fine.
Is there some way to setup the parents so autoprefixer can do its prefix detection?

**Comment by ai** [comment]
Can you show the full code of your plugin and PostCSS config?

**Comment by ksjogo** [comment]
Sure.
		```
		import * as postcss from 'postcss';
		import { unzip, uniq, repeat, split, size, trimChars, map, compose as c, join } from 'lodash/fp';
		import { forEach } from 'lodash'
		
		const append = (app: string | number) => (str: string) => `${str}${app}`
		const wrap = (delimiter: string | number) => (str: string) => `${delimiter}${str}${delimiter}`
		
		const rowLengths = (arr: any[][]) => uniq(map(size)(arr));
		const rowItems = (arr: any[][]) => size(uniq(arr[0]));
		
		const PLUGIN = 'postcss-videogrid-plugin'
		
		const plugin = postcss.plugin(PLUGIN, function () {
		    return function (root, rootresult) {
		        let gridIdx = 0;
		        root.walkAtRules('videogrid', (rule) => {
		            gridIdx++;
		            let unusedIdx = 1;
		            const unused = (str: string) => str.replace('_', () => `_${unusedIdx++}`)
		
		            const raw = c(
		                map(c(
		                    map(c(
		                        unused,
		                        append(gridIdx))),
		                    split(' '),
		                    trimChars(' "'))),
		                split('\n'))
		                (rule.params);
		            console.log(raw);
		
		            const RowL = rowLengths(raw);
		            const RowI = rowItems(raw);
		            const ColL = rowLengths(unzip(raw));
		
		            forEach({
		                Row: RowL,
		                Col: ColL
		            }, (vals, key) => {
		                if (vals.length != 1) {
		                    throw rule.error(`${key} lengths must be equal, found: ${vals}`, { plugin: PLUGIN })
		                }
		            });
		
		            const result = c(
		                append(`/\n${repeat(RowL[0])('1fr ')}`),
		                join('\n'),
		                map(c(
		                    append(' 1fr'),
		                    wrap('"'),
		                    join(' '))))
		                (raw)
		
		            rule = rule.replaceWith({
		                prop: 'grid-template',
		                value: result
		            }, {
		                    prop: 'width',
		                    value: 'calc(100% - 2 * var(--gap))',
		                    source: rule.source
		                }, {
		                    prop: 'max-width',
		                    value: `calc((100vh - 2 * var(--gap)) * ${RowI * 16} / ${ColL[0] * 9} - 2 * var(--gap))`,
		                    source: rule.source
		                }, {
		                    prop: 'grid-gap',
		                    value: 'var(--gap)',
		                    source: rule.source
		                }, {
		                    selector: '.cell:nth-child(1)',
		                    nodes: [postcss.decl({
		                        prop: 'grid-area',
		                        value: 'a1'
		                    })]
		                }, //more grid-area selectors would be spliced in here
		            );
		
		            console.log({ gridIdx, result });
		        });
		    };
		});
		
		export default plugin; 
		```
		
		```
		require('ts-node').register({
		  transpileOnly: true,
		  project: './tsconfig.node.json'
		});
		
		/* eslint-disable @typescript-eslint/no-var-requires */
		/* eslint-disable import/no-commonjs */
		module.exports = {
		  plugins: [
		    require('./.../videogrid.ts').default,
		    require('postcss-utilities'),
		    require('postcss-css-variables'),
		    require('postcss-nested'),
		    require('autoprefixer')({
		      grid: 'no-autoplace'
		    })
		  ]
		};
		
		```
		
		Would be called like this:
		```
		.grid-1 {
		  @videogrid "a";
		}
		/*
		.grid-2 {
		  @videogrid "a b";
		}
		
		.grid-3 {
		  @videogrid "a a b b"
		   "_ c c _";
		}/*
		```
		
		I have no clue if that is the best way, I am trying to streamline manually unrolled loops.

**Comment by ai** [comment]
You can’t use `nodes:` option. Since PostCSS will not change `parent` and do other magic.

Maybe we should add a warning for non-`[]` value in `nodes` :thinking:

**Comment by ai** [comment]
PostCSS now will prevent issues like this https://github.com/postcss/postcss/commit/3494ffc2e78635944d1e68169dc1def324d3e371

**Comment by ai** [comment]
Released in PostCSS 7.0.19
